enclave/src/lib.rs

```diff
@@ -47,6 +47,51 @@ fn check_chain_info(chain_info: *const u8, chain_info_len: usize) -> Option<Chai
     }
 }
 
+#[inline]
+fn unseal(sealed_log: &mut [u8]) -> Option<TxWithOutputs> {
+    if sealed_log.len() >= (std::u32::MAX as usize) {
+        return None;
+    }
+    let opt = unsafe {
+        SgxSealedData::<[u8]>::from_raw_sealed_data_t(
+            sealed_log.as_mut_ptr() as *mut sgx_sealed_data_t,
+            sealed_log.len() as u32,
+        )
+    };
+    let sealed_data = match opt {
+        Some(x) => x,
+        None => {
+            return None;
+        }
+    };
+    let result = sealed_data.unseal_data();
+    let unsealed_data = match result {
+        Ok(x) => x,
+        Err(_) => {
+            return None;
+        }
+    };
+    let otx = TxWithOutputs::decode(&mut unsealed_data.get_decrypt_txt());
+    // TODO: check decoded txid against unsealed_data.get_additional_txt?
+    match otx {
+        Ok(tx) => Some(tx),
+        _ => None,
+    }
+}
+
+#[inline]
```

Comment:
Do we want to make this function inline?

Reply:
https://github.com/crypto-com/chain-tx-enclave/pull/11#discussion_r310433345